src/scripts/components/UploadListItem.js

```diff
@@ -0,0 +1,567 @@
+`
+
+class UploadListItem extends Component<Props, Object> {
+  handleInputChange: (input: string, e: Object) => void
+  onPublishVideo: (e: Object) => void
+  onSaveData: (e: Object) => void
+  publishVideo: (publish: boolean) => void
+  saveData: (publish: boolean) => void
+  handleHeight: (e: Object) => string
+  toggleOpen: (e: Object) => void
+
+  formWrapperRef: Object
+
+  constructor (props: Props, formWrapperRef: Object) {
+    super(props)
+    const theVideo = this.props.video
+    this.state = {
+      open: false,
+      id: theVideo.id,
+      title: theVideo.title,
+      description: theVideo.description,
+      duration: theVideo.duration,
+      author: theVideo.author,
+      height: '0px'
+    }
+
+    this.formWrapperRef = formWrapperRef
+
+    this.handleHeight = this.handleHeight.bind(this)
+    this.toggleOpen = this.toggleOpen.bind(this)
+    this.onSaveData = this.onSaveData.bind(this)
+    this.onPublishVideo = this.onPublishVideo.bind(this)
+    this.publishVideo = this.publishVideo.bind(this)
+    this.saveData = this.saveData.bind(this)
+    this.handleInputChange = this.handleInputChange.bind(this)
+  }
+
+  handleInputChange (input: string, e: Object) {
+    this.setState({
+      [input]: e.target.value
+    })
+  }
+
+  onSaveData (e: Object) {
+    e.preventDefault()
+    if (this.props.isWalletSecured) {
+      this.saveData(false)
+    } else {
+      // If wallet not secure open the modal
+      this.props.checkUserWallet()
+    }
+  }
+
+  saveData (publish: false) {
+    const videoToSave = {
+      id: this.state.id,
+      title: this.state.title,
+      description: this.state.description,
+      author: this.props.user.name
+    }
+    this.props.saveVideoInfo(videoToSave)
+  }
+
+  onPublishVideo (e: Object) {
+    e.preventDefault()
+    const balance = Number(this.props.user.balances.PTI) // paratii.eth.web3.utils.fromWei(balance)
+    // FIXME we need to manage this globally and not hardcoded
+    const stakeAmount = 5
+    const stakeAmountWei = Number(
+      paratii.eth.web3.utils.toWei(stakeAmount + '')
+    )
+    if (balance < stakeAmountWei) {
+      this.props.notification(
+        {
+          title: 'Not enough tokens',
+          message: `You need at least ${stakeAmount} PTIs to make a stake.`
+        },
+        'error'
+      )
+    } else {
+      if (this.props.isWalletSecured) {
+        this.publishVideo(true)
+      } else {
+        // If wallet not secure open the modal for signup / login
+        this.props.checkUserWallet()
+      }
+    }
+  }
+
+  publishVideo (publish: false) {
+    this.props.openModal(MODAL.STAKE)
+  }
+
+  toggleOpen (e: Object) {
+    this.setState({
+      open: !this.state.open
+    })
+  }
+
+  handleHeight (e: Object) {
+    let height = '0px'
+    if (this.formWrapperRef && this.state.open) {
+      height = this.formWrapperRef.offsetHeight + 'px'
+    }
+    return height
+  }
+
+  render () {
+    const { video, videoId } = this.props
```

Comment:
This file and render method are huge. Could it be broken up?

`getStatusMessage` should be something like an `operator` or `util` and live outside of this component. Other parts of `render` can go into helper `renderXXX` methods. Will be easier to reason about this component later if this refactor is doen

Reply:
Agree, but we were in a hurry to finish this